Workout/SwimmingWorkout.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  SwimmingWorkout.swift
+//  Workout
+//
+//  Created by Marco Boschi on 04/03/2017.
+//  Copyright © 2017 Marco Boschi. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+import MBLibrary
+
+class SwimmingWorkout: Workout {
+	
+	override init(_ raw: HKWorkout, delegate del: WorkoutDelegate?) {
+		super.init(raw, delegate: del)
+		
+		if #available(iOS 10, *) {
+			self.addDetails()
+			self.addRequest(for: HKQuantityTypeIdentifier.distanceSwimming.getType()!, withUnit: .kilometer(), andTimeType: .ranged, searchingBy: .workout)
```

Comment:
I would say that pace is actually better than km/h. (but that might be my perspective only).
Otherwise, the details you mentioned are perfect.
One question about heart rate. Are you computing an average of the heart rate / min?